modules/zone/main.tf

```diff
@@ -10,10 +10,11 @@ terraform {
 locals {
   hosts_cidr_block      = cidrsubnet(var.zone_ipv4_cidr, 1, 1)
   hosts_ipv6_cidr_block = cidrsubnet(aws_vpc.main.ipv6_cidr_block, 8, 1)
+  zone_az               = coalesce(var.zone_az, var.zone.az)
```

Comment:
Would be nice if we could instead end up with a local 'zone' here, and use that everywhere below, instead of sometimes refering to zone_az below, and sometimes var.zone / var.zone.foo.